tests/unique_pointer.t.cpp

```diff
@@ -0,0 +1,23 @@
+#include "unique_pointer.hpp"
+#include <gtest/gtest.h>
+#include <algorithm>
+
+TEST(UniquePointer, creatingNew) {
+     UniquePointer<int> uniquePointer = make_unique<int>(5);
+     UniquePointer<int> uniquePointerOne = make_unique<int>(10);
+     EXPECT_EQ(*uniquePointer, 5);
+     EXPECT_EQ(*uniquePointerOne, 10);
+     uniquePointer = (std::move(uniquePointerOne));
+     EXPECT_EQ(*uniquePointer, 10);
+}
+
+
+TEST(UniquePointer, creatingUniquePointerArray) {
+    std::vector<UniquePointer<int>> uniquePointerArray;
```

Comment:
Also do it for a custom type S.

Put counters in constructor / destructor of S.

Reply:
done